packages/miniflare/src/plugins/assets/index.ts

```diff
@@ -239,10 +239,9 @@ const walk = async (dir: string) => {
 				 */
 
 				const [pathHash, contentHash] = await Promise.all([
-					hashPath(encodeFilePath(relativeFilepath, path.sep)),
-					hashPath(
-						encodeFilePath(filepath, path.sep) + filestat.mtimeMs.toString()
-					),
+					hashPath(normalizeFilePath(relativeFilepath)),
+					// used absolute filepath here so that changes to the enclosing asset folder will be registered
+					hashPath(filepath + filestat.mtimeMs.toString()),
```

Comment:
Note what goes into contentHash doesn't actually matter, as long as it is uniquely associated with that filepath and updates with file changes. Just removed the filepath encoding because it was unnecessary and because normalizeFilePath() expects a relative filepath now.